src/Infrastructure.Styles.Analyzer/RMSYN1102SeparateExitConditionsAnalyzerTools/SplitExpressionUtility.cs

```diff
@@ -0,0 +1,86 @@
+// This file is part of the re-motion Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+// 
+// re-motion Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+// 
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+// 
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Infrastructure.Styles.Analyzer.RMSYN1102SeparateExitConditionsAnalyzerTools
+{
+  public class SplitExpressionUtility
+  {
+    public static (SyntaxNode Left, SyntaxNode Right) SplitExpression (SyntaxNode node)
+    {
+      if (node.IsKind(SyntaxKind.LogicalOrExpression)) return SplitLogicalOrExpression(node);
+
+      if (node.IsKind(SyntaxKind.OrPattern)) return SplitOrPattern(node);
```

Comment:
unless it's a short argument check we don't use single-line if-statements to allow the reader's eye to take in one intention per line.